### PR TITLE
Update call sites using the national number pattern to do so through MatcherApi

### PR DESCRIPTION
**Pull request by keghani**
cl/154533233
b/37494614

---
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1749)

### CODE REVIEW

File: cpp/src/phonenumbers/matcher_api.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ class MatcherApi {
   // Returns whether the given national number (a string containing only decimal
   // digits) matches the national number pattern defined in the given
   // PhoneNumberDesc message.
-  virtual bool MatchesNationalNumber(const string& number,
+  virtual bool MatchNationalNumber(const string& number,
                                      const PhoneNumberDesc& number_desc,
```
Comment (lararennie) [inline cpp/src/phonenumbers/matcher_api.h]: Indentation off in line 41 and 41.

File: cpp/src/phonenumbers/phonenumberutil.cc [hunk 2]
```diff
@@ -34,11 +34,13 @@
 #include "phonenumbers/base/memory/singleton.h"
 #include "phonenumbers/default_logger.h"
 #include "phonenumbers/encoding_utils.h"
+#include "phonenumbers/matcher_api.h"
 #include "phonenumbers/metadata.h"
 #include "phonenumbers/normalize_utf8.h"
 #include "phonenumbers/phonemetadata.pb.h"
 #include "phonenumbers/phonenumber.h"
 #include "phonenumbers/phonenumber.pb.h"
+#include "phonenumbers/regex_based_matcher.h"
```
Comment (lararennie) [inline cpp/src/phonenumbers/phonenumberutil.cc]: It is unfortunate (I now notice) that we use regexp elsewhere in our filenames and regex here.... anyway!! :)

File: cpp/src/phonenumbers/regex_based_matcher.h [hunk 3]
```diff
@@ -38,7 +38,7 @@ class RegexBasedMatcher : public MatcherApi {
   RegexBasedMatcher();
   ~RegexBasedMatcher();
 
-  bool MatchesNationalNumber(const string& national_number,
+  bool MatchNationalNumber(const string& national_number,
                              const PhoneNumberDesc& number_desc,
```
Comment (lararennie) [inline cpp/src/phonenumbers/regex_based_matcher.h]: same here re indentation

File: cpp/src/phonenumbers/phonenumberutil.cc (hunk 2 above)
Comment (keghani) [inline cpp/src/phonenumbers/phonenumberutil.cc]: :)

File: cpp/src/phonenumbers/regex_based_matcher.h (hunk 3 above)
Comment (keghani) [inline cpp/src/phonenumbers/regex_based_matcher.h]: Done

File: cpp/src/phonenumbers/matcher_api.h (hunk 1 above)
Comment (keghani) [inline cpp/src/phonenumbers/matcher_api.h]: Done

**Pull request merged**